### PR TITLE
Eval temp

### LINKED ISSUE #28: raneem(evaluate-temperature)
A function that evaluates the safety of a temperature

you add  temperature and you get its safety level 
(for example: -10 celsius is going to be evaluated to "freezing")
(another example: 30c is going to be evaluated to "safe")

### PR DESCRIPTION
**Pull request by Raneemrami**
---
name: evaluate_temperature
about: evaluate the safety level of a temperature
---

it gives you the safety level of the added temperature, here is the function and its unittests. 
Examples:
Evaluates -10 Celsius to "Freezing"
Evaluates 30  Celsius to "Safe"
Evaluates 39  Celsius to "Normal"
issue link:
[#28](https://github.com/MIT-Emerging-Talent/ET6-foundations-group-31/issues/28)

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by Raneemrami** [comment]
this closes #28

File: solutions/evaluate_temperature.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    >>> evaluate_temperature(-15)
+    'Freezing'
+
+    Notes:
+    - Temperatures below -10°C and above 50°C are considered dangerous.
+    """
+
+    assert isinstance(celsius, int), "celsius is not an integer"
+    assert celsius < 50, "celsius is greater than 50"
```
Comment (Akan186) [inline solutions/evaluate_temperature.py]: I think this should be changed to: assert celsius <= 50, "celsius is greater than 50"

File: solutions/tests/test_evaluate_temperature.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    def test_freezing_temperature(self):
+        """It should evaluate -15 to 'Freezing'"""
+        self.assertEqual(evaluate_temperature(-15), "Freezing")
+
+    def test_not_an_integer(self):
+        """It should raise an assertion error if the argument is not an integer"""
+        with self.assertRaises(AssertionError):
+            evaluate_temperature(1.0)
+
+    def test_less_than_0(self):
```
Comment (Akan186) [inline solutions/tests/test_evaluate_temperature.py]: Your comment and the def line contradict each other. You should correct it to where they both match. Change the def line to: test_greater_than_50

**Review by Akan186 — CHANGES_REQUESTED** [review]
Fix these 2 changes and it'll be perfect.

File: solutions/tests/test_evaluate_temperature.py (hunk 2 above)
Comment (Raneemrami) [reply solutions/tests/test_evaluate_temperature.py]: I fixed them, is it good now?

**Comment by Raneemrami** [comment]
is there anything I can fix?

**Comment by Akan186** [comment]
> is there anything I can fix?

it's perfect. I'll approve and merge it now.

**Comment by Raneemrami** [comment]
Thank you

**Review by Akan186 — APPROVED** [review]
Perfect!
